tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,477 @@
+        test_cases = [
+            {
+                "args": (self.user, "Test title", "Example content"),
+                "expected_output": {
+                    "description": "Example content",
+                    "title": "Test title",
+                    "icon_url": Icons.user_verified
+                },
+                "send_result": True,
+                "send_raise": None
+            },
+            {
+                "args": (self.user, "Test title 1", "Example content 1", Icons.user_update),
+                "expected_output": {
+                    "description": "Example content 1",
+                    "title": "Test title 1",
+                    "icon_url": Icons.user_update
+                },
+                "send_result": False,
+                "send_raise": NotFound(AsyncMock(), AsyncMock())
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            expected = case["expected_output"]
+            send, send_raise = case["send_result"], case["send_raise"]
+
+            with self.subTest(args=args, expected=expected):
+                if send_raise:
+                    self.ctx.send.side_effect = send_raise
+
+                send_private_embed_mock.return_value = send
+
+                result = await notify_pardon(*args)
+
+                self.assertEqual(send, result)
+
+                embed = send_private_embed_mock.call_args[0][1]
+
+                self.assertEqual(embed.description, expected["description"])
+                self.assertEqual(embed.colour.value, PARDON_COLOR)
+                self.assertEqual(embed.author.name, expected["title"])
+                self.assertEqual(embed.author.icon_url, expected["icon_url"])
+
+                send_private_embed_mock.assert_awaited_once_with(args[0], embed)
+
+                self.ctx.send.reset_mock(side_effect=True)
+                send_private_embed_mock.reset_mock()
+
+    async def test_post_user(self):
+        """Test does `post_user` work correctly."""
+        test_cases = [
+            {
+                "args": (self.ctx, self.user),
+                "post_result": [
+                    {
+                        "id": 1234,
+                        "avatar": "test",
+                        "name": "Test",
+                        "discriminator": 1234,
+                        "roles": [
+                            1234,
+                            5678
+                        ],
+                        "in_guild": False
+                    }
+                ],
+                "raise_error": False,
+                "payload": {
+                    "avatar_hash": getattr(self.user, "avatar", 0),
+                    "discriminator": int(getattr(self.user, "discriminator", 0)),
+                    "id": self.user.id,
+                    "in_guild": False,
+                    "name": getattr(self.user, "name", "Name unknown"),
+                    "roles": []
+                }
```

Comment:
Don't just copy the `getattr` stuff. To properly test this, use different users - one which has the attributes and one which doesn't. Then, here you'd hard code the values you expect the payload to have.

Also want to point out this is *currently* another instance of test data being repeated across test cases.

Reply:
Is like for one should I use random `avatar_hash`, but for second 0, same for `discriminator` and `name`?

Reply:
Yes that is what I meant. It seems to have been implemented correctly.